### PR TITLE
Add custom CLJS support code

### LINKED ISSUE #5: Add static build helpers from JSXGraph
Some of the code from https://github.com/mentat-collective/jsxgraph.cljs/blob/main/dev/user.clj will be useful to share across the mentat projects, and for other folks looking to build interactive docs notebooks that produce custom JS.

### PR DESCRIPTION
**Pull request by sritchie**
- #10:

  - Adds `mentat.clerk-utils.css` with `add-css!`, `set-css!` and `reset-css!`
    functions for customizing the CSS served in Clerk's header

  - Adds `mentat.clerk-utils.docs` with `git-sha` and `git-dependency` for
    injecting git dependency instructions into a notebook

  - Adds `mentat.clerk-utils.docs` with `serve!`, `build!` and `halt!`
    variations that support custom CLJS compilation
    
  - Modifies the `clerk-utils` project to use this new style, vs the old
    `shadow-cljs.edn` approach.

### CODE REVIEW

File: src/mentat/clerk_utils/shadow.clj [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  [cljs-namespaces]
+  (let [entries (into ['nextjournal.clerk.static-app]
+                      cljs-namespaces)]
+    {:build-id build-id
+     :target :esm
+     :runtime :browser
+     :modules {:main {:entries entries}}
+     :output-dir output-dir
+     :build-options
+     {:ns-aliases '{nextjournal.devcards nextjournal.devcards-noop}}
```
Comment (mk) [inline src/mentat/clerk_utils/shadow.clj]: `nextjournal.devcards` is no longer a dep of clerk so this is now obsolete.

File: src/mentat/clerk_utils/build.clj [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+  "Like `clerk/serve!` but handles custom cljs. If `browse?` and `show` are true,
+  shows the entry at `show!` once it opens."
+  [{:keys [cljs-namespaces browse? index] :as opts}]
+  (when (seq cljs-namespaces)
+    (let [{:keys [js-path]} (shadow/watch! cljs-namespaces)]
+      (set-viewer-js! js-path)))
+  (try (clerk/serve! opts)
+       (finally
+         (when (and browse? index)
+           (Thread/sleep 500)
```
Comment (mk) [inline src/mentat/clerk_utils/build.clj]: Curious why the sleep is needed, something we should fix upstream maybe?

**Review by mk — COMMENTED** [review]
Looks good! Just noticed two tiny things.

File: src/mentat/clerk_utils/build.clj (hunk 2 above)
Comment (sritchie) [reply src/mentat/clerk_utils/build.clj]: looks like it's not, I've removed it. Good catch!

**Pull request merged**